net/respondd-module-airtime/src/ifaces.c

```diff
@@ -0,0 +1,64 @@
+#include <sys/socket.h>
+#include <linux/nl80211.h>
+#include <netlink/netlink.h>
+#include <netlink/genl/genl.h>
+#include <netlink/genl/ctrl.h>
+#include <net/if.h>
+#include <stdlib.h>
+
+#include "ifaces.h"
+
+static int iface_dump_handler(struct nl_msg *msg, struct iface_list **arg) {
+	struct nlattr *tb[NL80211_ATTR_MAX + 1];
+	struct genlmsghdr *gnlh = nlmsg_data(nlmsg_hdr(msg));
+	int wiphy;
+	struct iface_list **last_next;
+
+	nla_parse(tb, NL80211_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0), NULL);
+
+	wiphy = nla_get_u32(tb[NL80211_ATTR_WIPHY]);
+	for (last_next = arg; *last_next != NULL; last_next = &(*last_next)->next) {
+		if ((*last_next)->wiphy == wiphy)
+			goto abort;
+	}
+	*last_next = malloc(sizeof(**last_next));
+	(*last_next)->next = NULL;
+	(*last_next)->ifx = nla_get_u32(tb[NL80211_ATTR_IFINDEX]);
+	(*last_next)->wiphy = wiphy;
+
+abort:
+	return NL_SKIP;
+}
+
+struct iface_list *get_ifaces() {
```

Comment:
I'm not happy with these publicly visible functions in a dynamically linked module. Please add `__attribute__ ((visibility ("hidden")))`. (Also for *get_airtime()*)